src/bluesky/consolidators.py

```diff
@@ -88,8 +88,12 @@ def __init__(self, stream_resource: StreamResource, descriptor: EventDescriptor)
 
         # Find data shape and machine dtype; dtype_numpy, dtype_str take precedence if specified
         data_desc = descriptor["data_keys"][self.data_key]
-        self.datum_shape = tuple(data_desc["shape"])
-        self.datum_shape = self.datum_shape if self.datum_shape != (1,) else ()
+        self.datum_shape: tuple[int, ...] = tuple(data_desc["shape"])
```

Comment:
```suggestion
        self.datum_shape: tuple[int, ...] = tuple(data_desc["shape"]) or (1, )
```
I would not raise an error here, since it would prevent us from ingesting old data that may not have `datum_shape` properly set. I'd suggest just forcing it to be `(1,)`, and if it happens that the file has more dimensions, we can correct the shape later in Tiled once the assets are registered. You may instead opt to raising a warning here, so that the user could fix the ophyd config, but I'm not sure it's necessary.